core/modules/replica/IngestServerConnection.cc

```diff
@@ -0,0 +1,520 @@
+        }
+        _isPartitioned = _databaseInfo.partitionedTables.end() != find(
+            _databaseInfo.partitionedTables.begin(),
+            _databaseInfo.partitionedTables.end(),
+            _table);
+        if (not _isPartitioned) {
+            if (_databaseInfo.regularTables.end() != find(
+                _databaseInfo.regularTables.begin(),
+                _databaseInfo.regularTables.end(),
+                _table)) {
+                throw invalid_argument(
+                        "no such table '" + _table + "' in a scope of database '" +
+                        _databaseInfo.name + "'");
+            }
+        }
+
+        // The next test is for the partitioned tables, and it's meant to check if
+        // the chunk number is valid and it's allocated to this worker. The test will
+        // also ensure that the database is in the UNPUBLISHED state.
+
+        if (_isPartitioned) {
+
+            vector<ReplicaInfo> replicas;       // Chunk replicas at the current worker found
+                                                // among the unpublished databases only
+            bool const allDatabases = false;
+            bool const isPublished = false;
+
+            _serviceProvider->databaseServices()->findWorkerReplicas(
+                replicas,
+                _chunk,
+                _workerName,
+                _databaseInfo.family,
+                allDatabases,
+                isPublished
+            );
+            bool databaseIsFound = false;
+            for (auto&& replica: replicas) {
+                if (replica.database() == _databaseInfo.name) {
+                    databaseIsFound = true;
+                    break;
+                }
+            }
+            if (not databaseIsFound) {
+                throw invalid_argument(
+                        "chunk " + to_string(_chunk) + " of the UNPUBLISHED database '" +
+                        _databaseInfo.name + "' is not allocated to worker '" + _workerName + "'");
+            }
+        }
+                
+    } catch (DatabaseServicesNotFound const& ex) {
+        _illegalParameters("invalid transaction identifier");
+        return;
+    } catch (invalid_argument const& ex) {
+        _illegalParameters(ex.what());
+        return;
+    }
+    
+    // Create a temporary file
+
+    boost::system::error_code errCode;
+    
+    string const fileExt = ".csv";
+    string const pattern = _databaseInfo.name + "-" + _table + "-" + to_string(_chunk) + "-" +
+                           to_string(_transactionId) + "-%%%%" + fileExt;
+    fs::path const baseFileName = fs::unique_path(pattern, errCode);
+    if (errCode.value() != 0) {
+        _failed("failed to allocate the name for a temporary file, error: " + errCode.message());
+        return;
+    }
+
+    fs::path const filePath = fs::path(_workerInfo.loaderTmpDir) / baseFileName;
+    _fileName = filePath.string();
+    _file.open(_fileName, ofstream::out);
```

Comment:
This may be good thing to make a separate function and put it in util. For fs::unique_path to avoid using an existing file, wouldn't it need the full path in the pattern? Also, since the file is created after the name is generated, is there a tiny chance that another process (or thread) used the name to open file in that gap? It may be worth making multiple attempts if the file can't be opened due to a file already existing.

Reply:
and then you could unit test the tmp file code in util! 🙃